packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx

```diff
@@ -97,8 +108,10 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
 
   public render() {
     let rootClass = 'ms-DatePicker';
-    let { firstDayOfWeek, dateRangeType, strings, isMonthPickerVisible, isDayPickerVisible, autoNavigateOnSelection, showGoToToday, highlightCurrentMonth, navigationIcons } = this.props;
-    let { selectedDate, navigatedDate } = this.state;
+    let { firstDayOfWeek, dateRangeType, strings, showMonthPickerAsOverlay, autoNavigateOnSelection, showGoToToday, highlightCurrentMonth, navigationIcons } = this.props;
+    let { selectedDate, navigatedDate, isMonthPickerVisible, isDayPickerVisible } = this.state;
+    let onHeaderClick = showMonthPickerAsOverlay ? this._onHeaderClick : undefined;
```

Comment:
Yes, looks like headerClick is not the right thing to call it.  Changing to _onHeaderSelect to be more in-line with navigation naming conventions.